client/lib/menu-data/menu-data.js

```diff
@@ -411,7 +434,47 @@ MenuData.prototype.saveMenu = function( menu, callback ) {
 			this.change( { reset: true } );
 			this.emit( 'saved' );
 			callback && callback( null, parsedMenu );
-		} );
+		}
+	);
+}
+
+MenuData.prototype.saveMenu = function( menu, callback ) {
+	if ( ! menu ) {
+		menu = this.find( { id: this.lastChangedMenuID } );
+	}
+
+	if ( ! this.isValidMenu( menu ) ) {
+		callback && callback( new Error( 'Invalid menu' ) );
+		debug( 'saveMenu: fail' );
+		return;
+	}
+
+	menu = cloneDeep( menu );
+
+	if ( menu.id === this.getDefaultMenuId() ) {
+		return this.saveDefaultMenu();
+	}
+
+	this.restoreServerIDs( menu );
+
+	debug( 'saveMenu', menu );
+
+	const self = this;
```

Comment:
The functions instantiated below are arrow functions, thus context is preserved, thus the `self` hack shouldn't be necessary. Or did I miss something?